### PR TITLE
doc: relnotes: 3.5: Random subsys additional info

### PR DESCRIPTION
**Pull request by ceolin**
Add information about random subsys header renaming.

### CODE REVIEW

**Review by fabiobaltieri — DISMISSED** [review]

File: doc/releases/release-notes-3.5.rst [hunk 1]
```diff
@@ -802,6 +802,8 @@ Libraries / Subsystems
 
   * ``CONFIG_XOROSHIRO_RANDOM_GENERATOR``, deprecated a long time ago, is finally removed.
 
+  * Random API header ``rand32.h`` is deprecated in favor of ``random.h``.
```
[the remark this answers is not in the document]
Comment (ceolin) [reply doc/releases/release-notes-3.5.rst:805]: Yep, I was confuse about it tbh. Should the information be duplicated in both documents or only in the migration ?

[the remark this answers is not in the document]
Comment (jhedberg) [reply doc/releases/release-notes-3.5.rst:805]: According to the guidance (from @carlescufi ) it should only be in one place.

[the remark this answers is not in the document]
Comment (ceolin) [reply doc/releases/release-notes-3.5.rst:805]: ah ok, found it https://github.com/zephyrproject-rtos/zephyr/pull/63767

**Comment by ceolin** [comment]
@jhedberg @fabiobaltieri Moved the information to the migration doc. Please take a look if that is right please.

**Review by jhedberg — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**